### ISSUE TITLE
 thousand of warning due to bad UTF  using merged dictionaries  - at line:1 to 828525 (828525 warnings) #4097 

### ISSUE THREAD

**Issue by estatistics** [geany/geany-plugins#1392]
geany devs told me to report it here the bug. 

I have installed spellcheck. Cause i wanted to spellcheck both in greek and in english language, i joined two dictionaries together, a Greek one with an english one by simply adding one another below the other. adjusting any difference in number of words in the head. Nothing else. 

First was inserted the greek one and then the english one. the .dic file is a normal text files with a list of words and at its head has the number of words. Then adjusted aff files accordingly (normal txt files). Kate can normally open both files, almost immediately with correct iso code.

- However, geany prints 828525 lines of warnings in terminal before open the file. This means that it takes about ~1 minute to geany to open.  And also fail to detect the right iso which is 8859-7 (greek one -eastern europe).
- Moreover, when i set the iso code manually when it is already opened to correct one, fail to change the words to the right iso code.  I must set iso when opening file in more parameters in order to display the characters in the right manner.

geany reference:
[https://github.com/geany/geany/issues/4097](url)

**Comment by estatistics** [comment]
you can try your self  here the file: 

for 2 weeks: 
`https://files.fm/u/2sbfabdwwx`

**Comment by estatistics** [comment]
now listed as [1392](https://github.com/geany/geany-plugins/issues/1392)

Not at all! They remain to correct UTF of already opened file.

In spellcheck you can specify a dictionary. When you specify this, EVERYTIME that geany is opened, it prints 828525 because of the dictionary that i have defined in spellcheck. 

So, my solutions is either to remove/add dictionary each time AFTER the opening of geany. Howeve,r even then geany through the warnings. 

I CANNOT change the UTF coding of how SPEELCHECK uses the dictionaries.

So, it is another thing the UTF coding of a text file in geany AND different thing of how spellcheck/geany uses an added dictionary as an inside file. However, i think both bugs are relevant to geany and not to spellcheck plugin.

**Comment by techee** [comment]
@estatistics As I tried to tell you in https://github.com/geany/geany/issues/4097, you must convert the dictionary to the correct encoding, i.e. UTF-8. Since we are walking in circles, I did this for you:

https://www.dropbox.com/scl/fi/m7szrv676pw8nukrkfilb/el_GR.dic?rlkey=10mr9hubq5bmzkcyr37pc5ht8&dl=0

This is your dictionary converted to UTF-8 encoding - I haven't tested it but I guess it will work.

For your future reference, this is what I did in Geany:
1. Opened your file (it clearly opened in incorrect encoding as words weren't readable)
2. File->Reload As->West European->Greek (ISO 8895-7). After that, it's displayed correctly.
3. Document->Set Encoding->Unicode->Unicode (UTF-8). This converts the file to the right encoding.
4. Saved the file.

The only problem I see is the amount of warnings - one could be enough. I haven't checked if they come from the plugin or the spell check library.

But the fact that you used incorrect encoding is a problem on your side.
